src/hAPI/README.md

```diff
@@ -1,5 +1,54 @@
+* The Previous, Next, and Stop buttons can be used to navigate back and forth in the interaction, which always follows the pattern:
+![a state machine diagram explaining how the audio-haptic experience works](./images/haptics_state_machine.png){:height="250px"}
+NOTE:  this diagram will be changed slighly to indicate a starting state
+
+![differentiating an object, segment, and subsegment](./images/object_segments.png){:height="250px"}
+
+ For example, the clicking the "Next Button" during the TTS narration of a segment will immediately skip the TTS and move to the sonified experience of that segment. Clicking the "Next" button again will move to the guided tour of that segment, starting with the first subsegment. Clicking "Previous" during a TTS narration of the second segment will run a guided tour of the last subsegment of the first segment.
+
+## How does it work?  
+
+![the dataflow between the main browser script, the worker, and the 2DIY](./images/browser_arch.png)
+
+Data from the handler is sent to the ```hapi-utils.ts``` main script file.
+
+Code to play audio files is contained within the main script, while all haptic code runs inside a web worker. Since the audio and haptic experiences are intended to be run one after the other in a timed-fashion, there is back and forth communication between the main script and worker to keep track of state variables. For example, the ```doneWithAudio``` variable is set to true once an audio file has finished playing, sent to the worker via a ```postMessage()```, and  then acknowledged by the worker to ready the next haptic segment or object to be played. The code that handles the interactivity with the Previous, Next, and Stop buttons is also contained in the ```worker.ts``` file.
+
+## Haptic Loop
+
+Code for the haptic calculations is contained in the ```worker.ts``` webworker file. A web worker allows scripts to be executed in background threads, allowing it to perform tasks without interfering with the main script. Since the force calculations required for the 2DIY can be intensive, the web worker ensures that the performance of the main script is not hindered.
+ 
+The worker is instantiated in the ```hapi-utils.ts``` script by sending a preliminary message containing the  data points for the segments and objects contained in the image. The normalized coordinates returned from the handler are mapped into the 2DIY's frame of reference and then grouped into specific data types (segments, or objects) which are then utilized in functions called from the haptic loop to trace contours or move to certain positions when necessary.
+
+The worker is currently set up to infinitely loop, refreshing at a rate of 1 ms. However, stability issues occur when ensuring that this is maintained at all times. 
```

Comment:
Suggest linking back to the issue for this.

Reply:
I linked to #223.